test/suites/apparmor.sh

```diff
@@ -0,0 +1,117 @@
+test_apparmor() {
+  if [ ! -e /sys/module/apparmor/ ]; then
+    export TEST_UNMET_REQUIREMENT="missing AppArmor kernel support"
+    return
+  fi
+
+  ensure_import_testimage
+
+  echo "Create and start a test container"
+  lxc launch testimage c1
+
+  echo "==> Test /sys/kernel/* access"
+
+  echo "Verify access to /sys/kernel/config denied"
+  ! lxc exec c1 -- ls /sys/kernel/config/ || false
+
+  echo "Verify access to /sys/kernel/debug denied"
+  ! lxc exec c1 -- ls /sys/kernel/debug/ || false
+
+  echo "Verify access to /sys/kernel/tracing denied"
+  ! lxc exec c1 -- ls /sys/kernel/tracing/ || false
+
+  echo "Verify read access to /sys/kernel/security allowed"
+  lxc exec c1 -- ls /sys/kernel/security/
+
+  echo "Verify write access to /sys/kernel/security denied"
+  ! lxc exec c1 -- touch /sys/kernel/security/testfile || false
+
+  echo "Verify write access to /sys/kernel/security/apparmor allowed"
+  lxc exec c1 -- touch /sys/kernel/security/apparmor/.load
+
+  echo "==> Test /sys/firmware/efi/efivars/ access"
+
+  echo "Verify read access to /sys/firmware/efi/efivars/ denied"
+  ! lxc exec c1 -- ls /sys/firmware/efi/efivars/ || false
```

Comment:
This is the one dir I'm not 100% sure will be present even in the host. I don't know if Azure VMs are CSM or EFI based TBH.

This might need to be gated by the existence of that dir in the host.

Reply:
Locally with my EFI based host, I get a permission denied in the container. On Azure, we get:

```
2025-10-28T19:02:58.7592696Z + lxc exec c1 -- ls /sys/firmware/efi/efivars/
2025-10-28T19:02:58.7593488Z + timeout --foreground 120 /home/runner/go/bin/lxc exec c1 --verbose --force-local -- ls /sys/firmware/efi/efivars/
2025-10-28T19:02:58.7911153Z ls: /sys/firmware/efi/efivars/: No such file or directory
```